integration_tests/src/main/python/json_test.py

```diff
@@ -523,6 +523,42 @@ def test_from_json_struct_decimal():
             .select(f.from_json('a', 'struct<a:decimal>')),
         conf={"spark.rapids.sql.expression.JsonToStructs": True})
 
+@pytest.mark.parametrize('date_gen', [
+    # "yyyy-MM-dd"
+    "\"[ \t\xA0\u1680\u180e\u2000-\u200a\u202f\u205f\u3000]?[1-8]{1}[0-9]{3}-[0-3]{1,2}-[0-3]{1,2}[ \t\xA0\u1680\u180e\u2000-\u200a\u202f\u205f\u3000]?\"",
```

Comment:
Do we have any tests for JSON scan that would be similar to this? My concern is that this PR has updated some of JSONScan, with the tagging, but it does not look like it has updated the part that does the parsing. I am nervous that we are not keeping the two together and especially not with testing.

Reply:
We do now have some differences in heavier and testing between scan and from_json. There are some lessons learned from from_json that could now be applied to scans as well. I will file an issue to look at this.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/9750